widgets/ScrollingMessage.lua

```diff
@@ -0,0 +1,134 @@
+--- @type StdUi
+local StdUi = LibStub and LibStub('StdUi', true)
+if not StdUi then
+	return
+end
+
+local module, version = 'ScrollingMessage', 1
+if not StdUi:UpgradeNeeded(module, version) then
+	return
+end
+
+----------------------------------------------------
+--- ScrollingMessageFrame
+----------------------------------------------------
+
+StdUi.ScrollingMessageFrameMethods = {
+
+	AddMessage = function(self, ...)
+		-- must check whether we're max scrolled BEFORE we add the new message
+		local wasScrolledToMax = self:IsScrolledToMax()
+		-- parent class AddMessage
+		self:_AddMessage(...)
```

Comment:
Please don't use underscores in method names, if you need some internal name just name it AddMessageInternal

Reply:
Fixed locally, will push.